client/onboarding-prototype/steps/test/business-details.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import user from '@testing-library/user-event';
+import { mocked } from 'ts-jest/utils';
+
+/**
+ * Internal dependencies
+ */
+import BusinessDetails from '../business-details';
+import { OnboardingContextProvider } from '../../context';
+import { useBusinessTypes } from 'onboarding-experiment/hooks';
+import strings from '../../strings';
+
+jest.mock( 'onboarding-experiment/hooks', () => ( {
+	useBusinessTypes: jest.fn(),
+} ) );
+
+const countries = [
+	{
+		key: 'US',
+		name: 'United States',
+		types: [
+			{
+				key: 'individual',
+				name: 'Individual',
+				description: 'Individual description',
+				structures: [],
+			},
+			{
+				key: 'company',
+				name: 'Company',
+				description: 'Company description',
+				structures: [
+					{
+						key: 'sole_proprietorship',
+						name: 'Sole proprietorship',
+					},
+					{
+						key: 'single_member_llc',
+						name: 'Single member LLC',
+					},
+				],
+			},
+		],
+	},
+	{
+		key: 'FR',
+		name: 'France',
+		types: [
+			{
+				key: 'individual',
+				name: 'Individual',
+				description: 'Individual description',
+				structures: [],
+			},
+			{
+				key: 'company',
+				name: 'Company',
+				description: 'Company description',
+				structures: [],
+			},
+			{
+				key: 'non_profit',
+				name: 'Non-profit',
+				description: 'Non-profit description',
+				structures: [],
+			},
+		],
+	},
+];
+
+mocked( useBusinessTypes ).mockReturnValue( {
+	countries,
+	isLoading: false,
+} );
+
+describe( 'BusinessDetails', () => {
+	it( 'renders and updates fields data when they are changed', () => {
+		render(
+			<OnboardingContextProvider>
+				<BusinessDetails />
+			</OnboardingContextProvider>
+		);
+		const businessNameField = screen.getByLabelText(
+			strings.fields.business_name
+		);
+		const urlField = screen.getByLabelText( strings.fields.url );
+		const countryField = screen.getByText( strings.placeholders.country );
+
+		const mccField = screen.getByText( strings.placeholders.mcc );
```

Comment:
GH actions say 'mccField' is assigned a value but never used, it probably because of the todo below?
> // @ismaeldcom TODO: Add mcc field test

Reply:
Yep, we can remove that line, but I left it in purpose to make it clear that we need to use it.

Reply:
Gotcha! Let it be here, we'll address it in #4853